eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -223,6 +223,64 @@ def delete_enrollment(self, tenant: dict, data: dict | None = None) -> requests.
         return make_request(tenant, "DELETE", url=self.ENROLLMENT_URL, data=data)
 
 
+class PreEnrollmentAPIRequestMixin:
+    """Mixin class for the Pre Enrollments API request methods."""
+
+    PRE_ENROLLMENT_URL = f"{settings['EOX_CORE_API_BASE']}{reverse('eox-api:eox-api:edxapp-pre-enrollment')}"
+
+    def create_pre_enrollment(self, tenant: dict, data: dict) -> requests.Response:
+        """
+        Create a new pre-enrollment in a tenant.
+
+        Args:
+            tenant (dict): The tenant data.
+            data (dict): The pre-enrollment data.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "POST", url=self.PRE_ENROLLMENT_URL, data=data)
+
+    def get_pre_enrollment(self, tenant: dict, data: dict | None = None) -> requests.Response:
+        """
+        Get a pre-enrollment in a tenant by username or email.
+
+        Args:
+            tenant (dict): The tenant data.
+            data (dict, optional): The body data for the request.
+
+        Returns:
+            requests.Response: The response object.
+        """
+        return make_request(tenant, "GET", url=self.PRE_ENROLLMENT_URL, data=data)
```

Comment:
For the detail endpoint (GET and DELETE), should we pass the data using `params` instead of the `data` keyword? It looks like the endpoints accept both, but the conventional approach is to use `params`, so I think we should encourage that

Reply:
Yes, you're right!: https://github.com/eduNEXT/eox-core/commit/5d4c119d2cc1c356229cf6d7e4fdf17fef956c1c